### PR TITLE
Taxonomies

### PR DESCRIPTION
**Pull request by jesseeproductions**
- Adds all the taxomies to their own classes
- Changes link to the Parent Menu to be the Session admin listing
- Taxomies show in the admin listing for the respective post type.

Artifact:
-Adds Conference Schedule Parent Meeting to Admin Nav - https://cloudup.com/c0ZppDJXd0k

_Ref:_ [TEC-4882](https://theeventscalendar.atlassian.net/browse/TEC-4882)

[TEC-4882]: https://theeventscalendar.atlassian.net/browse/TEC-4882?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Conference/Admin/Menu.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ public function add_conference_schedule_menu() {
 			'Conference Schedule',
 			'Conference Schedule',
 			'read',
-			$this->menu_slug,
+			'edit.php?post_type=' . Plugin::SESSION_POSTTYPE,
```
Comment (bordoni) [inline src/Conference/Admin/Menu.php]: Why not use menu slug on this?

File: src/Conference/Admin/Menu.php [hunk 2]
```diff
@@ -67,8 +67,32 @@ public function organize_post_types() {
...
+		// Submenu for Locations Taxonomy.
+		add_submenu_page(
+			$this->menu_slug,
+			'Locations',
+			'Locations',
+			'read',
+			'edit-tags.php?taxonomy=' . Plugin::LOCATION_TAXONOMY . '&post_type=' . Plugin::SESSION_POSTTYPE
+		);
+		// Submenu for Tags Taxonomy.
+		add_submenu_page(
```
Comment (bordoni) [inline src/Conference/Admin/Menu.php]: Would love to save the hooks but not required.

File: src/Conference/Admin/Menu.php [hunk 3]
```diff
@@ -99,5 +132,14 @@ public function organize_post_types() {
 			'read',
 			'post-new.php?post_type=' . Plugin::SPONSOR_POSTTYPE
 		);
+		// Submenu for Sponsor Levels Taxonomy.
+		add_submenu_page(
+			$this->menu_slug,
```
Comment (bordoni) [inline src/Conference/Admin/Menu.php]: Would love to use `$this->get_menu_slug()` it allows for more flexibility later.

File: src/Conference/Taxonomies/Groups.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+			'labels'            => $group_labels,
+			'rewrite'           => [ 'slug' => 'session_group' ],
+			'query_var'         => 'session_group',
+			'hierarchical'      => true,
+			'public'            => true,
+			'show_ui'           => true,
+			'show_in_rest'      => true,
+			'show_admin_column' => true,
+			'rest_base'         => 'session_group',
+		] );
```
Comment (bordoni) [inline src/Conference/Taxonomies/Groups.php]: filter and save the returning object.

File: src/Conference/Taxonomies/Locations.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+			'labels'            => $location_labels,
+			'rewrite'           => [ 'slug' => 'location' ],
+			'query_var'         => 'location',
+			'hierarchical'      => true,
+			'public'            => true,
+			'show_ui'           => true,
+			'show_in_rest'      => true,
+			'show_admin_column' => true,
+			'rest_base'         => 'session_location',
+		] );
```
Comment (bordoni) [inline src/Conference/Taxonomies/Locations.php]: Filter and save retuning object.

File: src/Conference/Taxonomies/Provider.php [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+	public function register() {
+		// Register the SP on the container.
+		$this->container->singleton( 'tec.conference.taxonomies.provider', $this );
+		$this->container->singleton( Tracks::class, Tracks::class );
+		$this->container->singleton( Locations::class, Locations::class );
+		$this->container->singleton( Tags::class, Tags::class );
+		$this->container->singleton( Sponsor_Levels::class, Sponsor_Levels::class );
+		$this->container->singleton( Groups::class, Groups::class );
+
+		$this->add_actions();
```
Comment (bordoni) [inline src/Conference/Taxonomies/Provider.php:40]: Add the empty `add_filters`.

File: src/Conference/Taxonomies/Provider.php [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+	}
+
+	/**
+	 * Adds required actions for post types.
+	 *
+	 * @since TBD
+	 */
+	public function add_actions() {
```
Comment (bordoni) [inline src/Conference/Taxonomies/Provider.php]: Should we make it protected?

File: src/Conference/Taxonomies/Sponsor_Levels.php [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+			'labels'            => $sponsor_level_labels,
+			'rewrite'           => [ 'slug' => 'sponsor-level' ],
+			'query_var'         => 'sponsor-level',
+			'hierarchical'      => true,
+			'public'            => true,
+			'show_ui'           => true,
+			'show_in_rest'      => true,
+			'show_admin_column' => true,
+			'rest_base'         => 'session_sponsor_level',
+		] );
```
Comment (bordoni) [inline src/Conference/Taxonomies/Sponsor_Levels.php]: Filter args and save object.

File: src/Conference/Taxonomies/Tags.php [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+			'labels'            => $tag_labels,
+			'rewrite'           => [ 'slug' => 'session_tag' ],
+			'query_var'         => 'session_tag',
+			'hierarchical'      => false,
+			'public'            => true,
+			'show_ui'           => true,
+			'show_in_rest'      => true,
+			'show_admin_column' => true,
+			'rest_base'         => 'session_tag',
+		] );
```
Comment (bordoni) [inline src/Conference/Taxonomies/Tags.php]: Filter args save object.

**Review by bordoni — CHANGES_REQUESTED** [review]
small changes.

File: src/Conference/Admin/Menu.php (hunk 1 above)
Comment (jesseeproductions) [reply src/Conference/Admin/Menu.php]: I changed it to use get_menu_slug()

File: src/Conference/Admin/Menu.php (hunk 2 above)
Comment (jesseeproductions) [reply src/Conference/Admin/Menu.php]: I can do that, do you have an example of saving the hooks?

$locations = add_submenu_page();

Not sure what to do with it from here.

File: src/Conference/Taxonomies/Sponsor_Levels.php (hunk 8 above)
Comment (jesseeproductions) [reply src/Conference/Taxonomies/Sponsor_Levels.php]: I added the filter args. How would I save the object? I can get the return value and pass it to another filter?

File: src/Conference/Admin/Menu.php (hunk 2 above)
Comment (bordoni) [reply src/Conference/Admin/Menu.php]: Dont worry about we are good.

**Review by bordoni — APPROVED** [review]
Good to go.

**Pull request merged**